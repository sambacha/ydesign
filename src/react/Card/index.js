import React from 'react';
import PropTypes from 'prop-types';
import Box from '../Box';

const Card = React.forwardRef((props, ref) => <Box ref={ref} {...props} />);

Card.defaultProps = {
  position: 'relative',
  bg: 'white',
  border: 1,
  borderColor: 'light-gray',
  boxShadow: 2,
  p: 4,
};

Card.propTypes = {
  ...Box.propTypes,
  theme: PropTypes.object,
};

Card.displayName = 'Card';

export default Card;
