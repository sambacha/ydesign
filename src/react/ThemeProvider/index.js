import React from 'react';
import PropTypes from 'prop-types';
import { ThemeProvider as StyledThemeProvider } from 'styled-components';
import theme from '../theme';

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  );
};

ThemeProvider.propTypes = {
  theme: PropTypes.object,
};

ThemeProvider.defaultProps = {
  theme,
};

ThemeProvider.displayName = 'ThemeProvider';

export default ThemeProvider;
