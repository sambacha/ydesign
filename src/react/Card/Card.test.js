import Card from './'
import ThemeProvider from '../ThemeProvider'
import theme from '../theme'

describe('Card component sanity', () => {
  it('has name', () => {
    expect(Card.displayName).toBe('Card')
  })

  it('matches default snapshot', () => {
    const component = render(<Card />)
    expect(component).toMatchSnapshot()
  })

  it('matches themed snapshot', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Card />
      </ThemeProvider>
    )
    expect(component).toMatchSnapshot()
  })
})
