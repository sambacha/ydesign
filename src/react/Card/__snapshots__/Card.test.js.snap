// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component sanity matches default snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  background-color: #fff;
  border: 1px solid transparent;
  border-color: #eee;
  box-shadow: 0px 8px 16px rgba(0,0,0,0.1);
  padding: 32px;
}

<div
  class="c0"
/>
`;

exports[`Card component sanity matches themed snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  position: relative;
  background-color: #fff;
  border: 1px solid transparent;
  border-color: #eee;
  box-shadow: 0px 8px 16px rgba(0,0,0,0.1);
  padding: 32px;
}

<div
  class="c0"
/>
`;
